packages/api/src/useInfiniteQuery.ts

```diff
@@ -15,15 +15,18 @@ import { getQueryKeys } from './utils';
 
 const useInfiniteQuery = <T extends TSocketPaginateableEndpointNames>(
     name: T,
-    ...props: TSocketAcceptableProps<T, true, 'useInfiniteQuery'>
+    ...props: [...TSocketAcceptableProps<T, true, 'useInfiniteQuery'>, { isPaginated: boolean }?]
 ) => {
```

Comment:
Please let me know if this is a good way to type this hook for expanding its logic

Reply:
Can we not patch useInfiniteQuery, because the hook is for paginated call. I dont think its a good idea to make it not paginated. Better have 2 different hooks like `useTransactions` and `usePaginatedTransactions` IMO

Reply:
You mean having `useAllTransactions` and `usePaginatedTransactions`?

Reply:
Makes sense